File: app/src/main/java/ru/mamykin/foreignbooksreader/views/BookDetailsView.java
package ru.mamykin.foreignbooksreader.views;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface BookDetailsView extends MvpView {
    void openBook(int bookId);

    void showTitle(String title);

    void setHomeEnabled(boolean enabled);

    void showBookName(String name);

    void showBookAuthor(String author);

    void showBookPath(String path);

    void showBookCurrentPage(String currentPage);

    void showBookGenre(String genre);

    void showBookOriginalLang(String lang);

    void showBookCreatedDate(String date);
}

File: app/src/main/java/ru/mamykin/foboreader/presentation/bookdetails/BookDetailsView.kt
package ru.mamykin.foboreader.presentation.bookdetails

import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType
import ru.mamykin.foboreader.core.ui.BaseView
import java.util.*

@StateStrategyType(AddToEndSingleStrategy::class)
interface BookDetailsView : BaseView {

    fun showTitle(title: String)

    fun setHomeEnabled(enabled: Boolean)

    fun showBookName(name: String)

    fun showBookAuthor(author: String)

    fun showBookPath(path: String)

    fun showBookCurrentPage(currentPage: String)

    fun showBookGenre(genre: String)

    fun showBookOriginalLang(lang: String)

    fun showBookCreatedDate(date: Date)
}